transitions/core.py

```diff
@@ -282,11 +282,22 @@ def add_states(self, states, on_enter=None, on_exit=None):
                 entered. Only valid if first argument is string.
             on_exit (string or list): callbacks to trigger when the state is
                 exited. Only valid if first argument is string.
+
+        Notes:
+            if a model is provided and on_enter or on_exit is None, the model will be scanned for
+            methods matching the patern on_enter_<state>, on_exit_<state>
         """
+
         states = listify(states)
         for state in states:
             if isinstance(state, str):
-                state = State(state, on_enter=on_enter, on_exit=on_exit)
+                enter_fn = on_enter
+                if enter_fn is None and self != self.model and hasattr(self.model, 'on_enter_'+state):
```

Comment:
Given the way the library is structured (i.e states can have multiple callbacks), that makes sense.